src/dbnode/persist/types.go

```diff
@@ -40,10 +40,19 @@ type DataFn func(id ident.ID, tags ident.Tags, segment ts.Segment, checksum uint
 // blocks for a (shard, blockStart) combination.
 type DataCloser func() error
 
+// DeferCloser returns a DeferredCloser that persists the data checkpoint file when called.
+type DeferCloser func(cleanup func()) (DeferredCloser, error)
```

Comment:
Can we just make this:
```go
type DeferCloser func() (DataCloser, error)
```

So we can reuse the same type signature between the previous "Close" and the new "Close"?